src/physics/interpolatedCollision.js

```diff
@@ -0,0 +1,225 @@
+// circle 2 sweeps from q1 to q2 with radius going from s1 to s2
+export function sweepCircleVsSweepCircle ( p1 : Vec2D, r1 : number, p2 : Vec2D, r2 : number, q1 : Vec2D, s1 : number, q2 : Vec2D, s2 : number ) : null | Vec2D {
+  if (euclideanDist(p1,q1) < (r1 + s1)) {
+    return new Vec2D (0.5 * p1.x + 0.5 * q1.x, 0.5 * p1.y + 0.5 * q1.y);
+  }
+  else {
+    const u = p1.x + q2.x - p2.x - q1.x;
+    const v = p1.y + q2.y - p2.y - q1.y;
+    const w = r1   + s1   - r2   - s2  ;
+    const a0 = Math.pow(p1.x - q1.x, 2) + Math.pow(p1.y - q1.y, 2) - Math.pow(r1 + s1, 2);
+    const a1 = -2 * ((p1.x - q1.x) * u + (p1.y - q1.y) * v - (r1 + s1) * w);
+    const a2 = Math.pow(u, 2) + Math.pow(v, 2) - Math.pow(w, 2);
+    const t1 = solveQuadraticEquation(a0, a1, a2);
+
+    let t = null;
+    if (t1 !== null && !isNaN(t1)) {
+      const t2 = a0 / (a2 * t1);
+      if (t1 < 0 || t1 > 1) {
+        if (t2 < 0 || t2 > 1 || isNaN(t2)) {
+          t = null;
+        }
+        else {
+          t = t2;
+        }
+      }
+      else {
+        if (t2 < 0 || t2 > 1 || isNaN(t2)) {
+          t = t1;
+        }
+        else {
+          t = Math.min(t1,t2);
+        }
+      }
+    }
+
+    if (t === null) {
+      return null;
+    }
+    else {
+      const r = (1 - t) * r1 + t * r2;
+      const s = (1 - t) * s1 + t * s2;
+      const p = new Vec2D ( (1-t) * p1.x + t * p2.x, (1-t) * p1.y + t * p2.y);
+      const q = new Vec2D ( (1-t) * q1.x + t * q2.x, (1-t) * q1.y + t * q2.y);
+      const wp = s/(r+s);
+      const wq = r/(r+s);
+      return new Vec2D ( wp * p.x + wq * q.x, wp * p.y + wq * q.y );
+    }
+  }
+}
+
+// computes the first point of collision between:
+//  - sweeping circle, going from p1 with radius r1 to p2 with radius r2
+//  - fixed AABB with bottom left point bl and top right point tr
+export function sweepCircleVsAABB ( p1 : Vec2D, r1 : number, p2 : Vec2D, r2 : number, bl : Vec2D, tr : Vec2D ) : null | Vec2D {
+  const br = new Vec2D( tr.x, bl.y );
+  const tl = new Vec2D( bl.x, tr.y );
+  if (distanceToPolygon(p1,[bl,br,tr,tl]) <= r1) {
+    return p1;
+  }
+  else if (   (p1.x + r1 < bl.x && p2.x + r2 < bl.x)
+           || (p1.x - r1 > tr.x && p2.x - r2 > tr.x)
+           || (p1.y + r1 < bl.y && p2.y + r2 < bl.y)
+           || (p1.y - r1 > tr.y && p2.y - r2 > tr.y) ) {
+    return null;
+  }
+  else {
+    let checks;
+    if (p1.x <= bl.x) {
+      if (p1.y <= bl.y) { // bottom left corner
+        checks = [ { case : "corner", corner : bl, p1 : p1, p2 : p2, r1 : r1, r2 : r2 }
+                 , { case : "corner", corner : tl, p1 : p1, p2 : p2, r1 : r1, r2 : r2 }
+                 , { case : "corner", corner : br, p1 : p1, p2 : p2, r1 : r1, r2 : r2 }
+                 , { case : "line"  , line1 : vLineThrough(bl), line2 : [add(p1, new Vec2D(r1,0)), add(p2, new Vec2D(r2,0))] }
+                 , { case : "line"  , line1 : hLineThrough(bl), line2 : [add(p1, new Vec2D(0,r1)), add(p2, new Vec2D(0,r2))] }
+                 ];
```

Comment:
I am only using the beginning circle position to figure out which corners/edges need to be checked against. The case analysis can be refined by using the end circle position to weed out possible cases.